wifiphisher/common/interfaces.py

```diff
@@ -633,12 +629,54 @@ def on_exit(self):
         :type self: NetworkManager
         :return: None
         :rtype: None
+        ..note: The cards in _exclude_shutdown will not set to the original mac address
+                since these cards are not changed the mac addresses by the program.
         """
 
         for interface in self._active:
-            mac_address = self._name_to_object[interface].original_mac_address
+            if interface not in self._exclude_shutdown:
+                adapter = self._name_to_object[interface]
+                mac_address = adapter.original_mac_address
+                self.set_interface_mac(interface, mac_address)
 
-            self.set_interface_mac(interface, mac_address)
+
+def is_managed_by_network_manager(interface_name):
+    """
+    Check if the interface is managed by NetworkManager
+
+    :param network_adapter: A NetworkAdapter object
+    :type interface_name: NetworkAdapter
+    :return if managed by NetworkManager return True
+    :rtype: bool
+    .. note: When the NetworkManager service is not running, using bus.get_object
+        will raise the exception. It's safe to pass this exception since when
+        NetworkManger doesn't run, the manage property will be un-managed.
+
+        We first create the network_manager_proxy first, and use it to get the
+        network_manager object that implements the interface NM_MANAGER_INTERFACE_PATH.
+        This network_manager object can then get all the assoicated devices, and we can
+        uses these devices' paths to get the device objects. After finding the target
+        device object we can then check if this device is managed by NetworkManager or not.
+    """
+
+    bus = dbus.SystemBus()
```

Comment:
@blackHatMonkey  thanks for the guidance : ) I write three test cases for the three conditions you have mentioned. For the first one you mention, since I pass the Exception in the function : `is_managed_by_network_manager` so it will not raise exception. So I give the interface which is managed by default as function argument but the function should return the `unmanaged` since the `NetworkManager` is not running.